python/Ganga/Runtime/GPIexport.py

```diff
@@ -18,6 +18,31 @@
 from Ganga.GPIDev.Base.Objects import GangaObject
 from inspect import isclass
 
+def _addToInterface(interface, name, _object):
+
+    if isType(_object, GangaObject):
+        setattr(interface, name, addProxy(_object))
+    elif isclass(_object) and issubclass(_object, GangaObject):
+        setattr(interface, name, getProxyClass(_object))
+    else:
+        setattr(interface, name, _object)
+
+def exportToInterface(myInterface, name, _object, doc_section, docstring=None):
```

Comment:
I'm not promoting we _should_ get it via the GPI. I'm promoting that we get it via the methods in ```python/Ganga/Core/GangaRepository/__init__.py```
For the moment we abuse the GPI because it's not clear what's going on. I would prefer to have a simple way of accessing GPI objects rather than doing something complex to regenerate them as we need them (at the least this is wasteful).

I agree searching the interface ```__dict__``` is very lazy and to be avoided but it's done in the Proxy layer atm to avoid having to make changes to the Plugins and the Proxy in the same PR.
I'm happy to put together a pull request which removes the lookups against the interface ```__dict__``` in fact I'll open an issue for this exact thing.